lib/src/shortest_path.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// Returns the shortest path from [start] to [target] given the directed
+/// edges of a graph provided by [edges].
+///
+/// If [start] `==` [target], an empty [List] is returned and [edges] is never
+/// called.
+///
+/// [start], [target] and all values returned by [edges] must not be `null`.
+/// If asserts are enabled, an [AssertionError] is raised if these conditions
+/// are not met. If asserts are not enabled, violations result in undefined
+/// behavior.
+///
+/// This function assumes [T] implements correct and efficient `==` and
+/// `hashCode`.
+List<T> shortestPath<T>(T start, T target, Iterable<T> Function(T) edges) =>
```

Comment:
I think we should try to match the signature of existing methods.

```dart
List<T> shortestPath<K,T>(T start, T, target, K Function(V) key, Iterable<T> Function(T) edges)
```

Here `K` needs to have correct and efficient `==` and `hashCode`.

Or else we need ton consider changing the existing methods to

```dart
List<List<T>> stronglyConnectedComponents(Iterable<T> startNodes, Iterable<T> Function(T) edges)
```

Reply:
PTAL 🙄